### PR TITLE
refactor: 💡 [JIRA:HCPSDKFIORIUIKIT-2880] ActionItems Refactor

### PR DESCRIPTION
**Pull request by hengyi-zhang**

### CODE REVIEW

File: Sources/FioriSwiftUICore/_FioriStyles/ActionItemsStyle.fiori.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        HStack(spacing: 0) {
+            ForEach(configuration.actionItems ?? [], id: \.id) { element in
+                Button {
+                    configuration.didSelectActivityItem?(element)
+                } label: {
+                    element.icon
+                        .font(.fiori(forTextStyle: .body).weight(.light))
+                        .imageScale(.large)
+                        .foregroundColor(.preferredColor(.tintColor))
+                        .frame(width: 44, height: 44, alignment: .center)
```
Comment (JunSong-SH) [inline Sources/FioriSwiftUICore/_FioriStyles/ActionItemsStyle.fiori.swift:19]: Here, I am not sure if the icon should support customization or not. For now, I think the color always 'blue' here and consumer don't have chance to change it. 
So, maybe, we can add a bool flag, isCustom = false, in ActivityItemDataType and don't set those font, foregroundColor something like that when isCustom = true. In additional, add new constructer, public init(type: ActivityItemType, data: String? = nil, isCustom: Bool), for API backward compatibility and let consumer have a chance to set a customized icon.
@xiaoyu0722 @dyongxu do you think we need support the customization?

Comment (dyongxu) [reply Sources/FioriSwiftUICore/_FioriStyles/ActionItemsStyle.fiori.swift:19]: I think all elements in our UI (including icon) should allow customization, i.e. App developer can override the font/color etc if their App Designer prefers another theming. So we should have a way to allow this.

**Review by JunSong-SH — APPROVED** [review]
LGTM!

File: Apps/Examples/Examples/FioriSwiftUICore/ActionItems/ActionItemsExample.swift [hunk 2]
```diff
@@ -0,0 +1,141 @@
+import FioriSwiftUICore
+import SwiftUI
+
+struct ActionItemsExample: View {
```
Comment (xiaoyu0722) [inline Apps/Examples/Examples/FioriSwiftUICore/ActionItems/ActionItemsExample.swift:4]: `ActionItems` not `ActivityItem`?
I'm not sure if we need an individual example for this sub-component. Or should put it into `ActivityItem` category.

File: Sources/FioriSwiftUICore/Views/ActionItems/ActionItemsBuilder.swift [hunk 3]
```diff
@@ -0,0 +1,113 @@
+import SwiftUI
+
+/// :nodoc:
+public protocol ActionItemsList: View, _ViewEmptyChecking {
+    associatedtype V: View
+    var count: Int { get }
+    func view(at index: Int) -> V
+}
+
+struct ActionItemsListStack: ActionItemsList {
```
Comment (xiaoyu0722) [inline Sources/FioriSwiftUICore/Views/ActionItems/ActionItemsBuilder.swift:10]: Seems same with `IndexedViewContainer`, you can use `struct ActionItemsListStack: IndexedViewContainer` directly.

File: Sources/FioriSwiftUICore/_ComponentProtocols/BaseComponentProtocols.swift [hunk 4]
```diff
@@ -467,3 +467,17 @@ protocol _ProgressComponent {
...
+///
+/// ## Usage
+/// ```swift
+/// ActionItems(actionItems: [.init(type: .phone), .init(type: .email), .init(type: .message), .init(type: .videoCall), .init(type: .detail)]) { dataType in
+///     print("\(dataType)")
+/// }
+/// ```
+// sourcery: BaseComponent
+protocol _ActionItemsComponent {
+    // sourcery: resultBuilder.name = @ActionItemsBuilder, resultBuilder.backingComponent = ActionItemsListStack
```
Comment (xiaoyu0722) [inline Sources/FioriSwiftUICore/_ComponentProtocols/BaseComponentProtocols.swift:481]: Try below annotation.
```swift
// sourcery: resultBuilder.name = @IndexedViewBuilder
// sourcery: resultBuilder.backingComponent = ActionItemsListStack
// sourcery: resultBuilder.returnType = any IndexedViewContainer
```

**Review by xiaoyu0722 — CHANGES_REQUESTED** [review]

File: Sources/FioriSwiftUICore/Views/ActionItems/ActionItemsBuilder.swift (hunk 3 above)
Comment (hengyi-zhang) [reply Sources/FioriSwiftUICore/Views/ActionItems/ActionItemsBuilder.swift:10]: IndexedViewContainer doesn't conform View protocol

File: Sources/FioriSwiftUICore/Views/ActionItems/ActionItemsBuilder.swift [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        var temp: [any View] = []
+        for element in actionItemsData {
+            let item = Button {
+                element.didSelectActivityItem?()
+            } label: {
+                element.icon
+                    .font(.fiori(forTextStyle: .body).weight(.light))
+                    .imageScale(.large)
+                    .foregroundColor(.preferredColor(.tintColor))
+                    .frame(width: 44, height: 44, alignment: .center)
```
[the remark this answers is not in the document]
Comment (hengyi-zhang) [reply Sources/FioriSwiftUICore/Views/ActionItems/ActionItemsBuilder.swift:21]: fixed

File: Apps/Examples/Examples/FioriSwiftUICore/ActionItems/ActionItemsExample.swift (hunk 2 above)
Comment (hengyi-zhang) [reply Apps/Examples/Examples/FioriSwiftUICore/ActionItems/ActionItemsExample.swift:4]: ActionItems not ActivityItem！

File: Sources/FioriSwiftUICore/_FioriStyles/ActionItemsStyle.fiori.swift [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import FioriThemeManager
+import Foundation
+import SwiftUI
+
+// Base Layout style
+public struct ActionItemsBaseStyle: ActionItemsStyle {
+    @ViewBuilder
+    public func makeBody(_ configuration: ActionItemsConfiguration) -> some View {
+        // Add default layout here
+        configuration.actionItems
```
[the remark this answers is not in the document]
Comment (hengyi-zhang) [reply Sources/FioriSwiftUICore/_FioriStyles/ActionItemsStyle.fiori.swift:10]: configuration.actionItems should be treated as one single View.

**Review by xiaoyu0722 — APPROVED** [review]

File: Sources/FioriSwiftUICore/_FioriStyles/ActionItemsStyle.fiori.swift (hunk 1 above)
Comment (hengyi-zhang) [reply Sources/FioriSwiftUICore/_FioriStyles/ActionItemsStyle.fiori.swift]: Developer can customize font and foregroundColor through .actionItemsStyle modifier, like in the example.

**Pull request merged**